crates/net/discv5/src/enr.rs

```diff
@@ -0,0 +1,113 @@
+//! Interface between node identification on protocol version 5 and 4. Specifically, between types
+//! [`discv5::enr::NodeId`] and [`PeerId`].
+
+use discv5::enr::{CombinedPublicKey, Enr, EnrPublicKey, NodeId};
+use reth_primitives::{id2pk, pk2id, PeerId};
+use secp256k1::{PublicKey, SecretKey};
+
+/// Extracts a [`CombinedPublicKey::Secp256k1`] from a [`discv5::Enr`] and converts it to a
+/// [`PeerId`]. Note: conversion from discv5 ID to discv4 ID is not possible.
+pub fn enr_to_discv4_id(enr: &discv5::Enr) -> PeerId {
```

Comment:
this will panic for non-secp256k1?

Reply:
yes but that is caught already by discv5 on decoding level, the key won't decode if it's another type. so here in the reth code, it's infallible.

https://github.com/sigp/discv5/blob/6707bcf3da350ff273208bb38dc29acc4cc4850c/src/rpc.rs#L459-L460
https://github.com/sigp/enr/blob/83cd98990e7f191b310addb202869994a8f59c96/src/keys/k256_key.rs#L56-L60

Reply:
then this function should be pub crate

Reply:
I'll add a panic section